File: 0007-reverse-integer/0007-reverse-integer.py
class Solution(object):
    def reverse(self, x):
        if x<0:
            string=str(-x)
            rev=int(string[::-1])
            rev=-rev
            if rev < -(2**31) or rev > 2**31 - 1:
                return 0
            return rev

        else:
            string=str(x)
            rev=int(string[::-1])
            if rev < -(2**31) or rev > 2**31 - 1:
                return 0
            return rev
        

File: 0007-reverse-integer/0007-reverse-integer.cpp
#include <algorithm>
#include <string>
#include <climits>
using namespace std;

class Solution {
public:
    int reverse(int x) {
        bool negative = false;

        if (x < 0) {
            negative = true;
        }

        string str = to_string(x);

        // Remove '-' before reversing
        if (negative) {
            str = str.substr(1);
        }

        std::reverse(str.begin(), str.end());

        long long result = stoll(str);

        if (negative) {
            result = -result;
        }

        // Check 32-bit integer range
        if (result > INT_MAX || result < INT_MIN) {
            return 0;
        }

        return (int)result;
    }
};